plugin/generator.go

```diff
@@ -19,26 +19,36 @@ import (
 	"github.com/docker/docker/client"
 )
 
-var proxyServiceTasks bool
-var caddyNetworks map[string]bool
+var labelPrefixFlag string
+var proxyServiceTasksFlag bool
+var isTrue = regexp.MustCompile("^(true|True|TRUE|yes|Yes|YES|1)$")
 
 func init() {
-	flag.BoolVar(&proxyServiceTasks, "proxy-service-tasks", false, "Proxy to service tasks instead of VIP")
+	flag.StringVar(&labelPrefixFlag, "docker-label-prefix", "caddy", "Prefix for Docker labels")
+	flag.BoolVar(&proxyServiceTasksFlag, "docker-proxy-service-tasks", false, "Proxy to service tasks instead of VIP")
 }
 
-
-func getCaddyLabelPrefix() string {
+func getLabelPrefix() string {
 	if val := os.Getenv("CADDY_DOCKER_LABEL_PREFIX"); val != "" {
```

Comment:
This could probably be executed only once, inside Init. I don't think the environment variables can change after the process started.